indico/modules/events/util.py

```diff
@@ -509,3 +519,74 @@ def serialize_event_for_ical(event, detail_level):
     if detail_level == 'contributions':
         data['contributions'] = [serialize_contribution_for_ical(c) for c in event.contributions]
     return data
+
+
+def get_field_values(form_data):
+    """Split the form fields between custom and static"""
+    fields = {x: form_data[x] for x in form_data.iterkeys() if not x.startswith('custom_')}
+    custom_fields = {x: form_data[x] for x in form_data.iterkeys() if x.startswith('custom_')}
+    return fields, custom_fields
+
+
+def set_custom_fields(obj, custom_fields_data):
+    changes = {}
+    for custom_field_name, custom_field_value in custom_fields_data.iteritems():
+        custom_field_id = int(custom_field_name[7:])  # Remove the 'custom_' part
+        old_value = obj.set_custom_field(custom_field_id, custom_field_value)
+        if old_value != custom_field_value:
+            changes[custom_field_name] = (old_value, custom_field_value)
+    return changes
+
+
+class ZipGeneratorMixin:
+    """Mixin for RHs that generate zip with files"""
+
+    def _adjust_path_length(self, segments):
+        """
+        Shorten the path length to < 260 chars.
+
+        Windows' built-in ZIP tool doesn't like files whose
+        total path exceeds ~260 chars. Here we progressively
+        shorten the total until it matches that constraint.
+        """
+        result = []
+        total_len = sum(len(seg) for seg in segments) + len(segments) - 1
+        excess = (total_len - 255) if total_len > 255 else 0
+
+        for seg in reversed(segments):
+            fname, ext = path.splitext(seg)
+            cut = min(excess, (len(fname) - 10) if len(fname) > 14 else 0)
+            if cut:
+                excess -= cut
+                fname = fname[:-cut]
+            result.append(fname + ext)
+
+        return reversed(result)
+
+    def _iter_items(self, files_holder):
+        for f in files_holder:
+            yield f
+
+    def _generate_zip_file(self, files_holder, name_prefix='material', name_suffix=None):
```

Comment:
It's not always a list, it can be any object that contains an object that cont.... that contains a list of files :stuck_out_tongue: